pkg/resources/stream_on_table.go

```diff
@@ -0,0 +1,307 @@
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInShowWithMapping("mode", func(x any) any {
+			return x.(string) == string(sdk.StreamModeAppendOnly)
+		}),
+		Description: booleanStringFieldDescription("Specifies whether this is an append-only stream."),
+	},
+	"show_initial_rows": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		Default:          BooleanDefault,
+		ValidateDiagFunc: validateBooleanString,
+		Description:      booleanStringFieldDescription("Specifies whether to return all existing rows in the source table as row inserts the first time the stream is consumed."),
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the stream.",
+	},
+	AtAttributeName:     schemas.AtSchema,
+	BeforeAttributeName: schemas.BeforeSchema,
+	ShowOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `SHOW STREAMS` for the given stream.",
+		Elem: &schema.Resource{
+			Schema: schemas.ShowStreamSchema,
+		},
+	},
+	DescribeOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `DESCRIBE STREAM` for the given stream.",
+		Elem: &schema.Resource{
+			Schema: schemas.DescribeStreamSchema,
+		},
+	},
+	FullyQualifiedNameAttributeName: schemas.FullyQualifiedNameSchema,
+}
+
+func StreamOnTable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateStreamOnTable(false),
+		ReadContext:   ReadStreamOnTable(true),
+		UpdateContext: UpdateStreamOnTable,
+		DeleteContext: DeleteStreamContext,
+		Description:   "Resource used to manage streams on tables. For more information, check [stream documentation](https://docs.snowflake.com/en/sql-reference/sql/create-stream).",
+
+		CustomizeDiff: customdiff.All(
+			ComputedIfAnyAttributeChanged(streamOnTableSchema, ShowOutputAttributeName, "table", "append_only", "comment"),
+			ComputedIfAnyAttributeChanged(streamOnTableSchema, DescribeOutputAttributeName, "table", "append_only", "comment"),
+		),
+
+		Schema: streamOnTableSchema,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportStreamOnTable,
+		},
+	}
+}
+
+func ImportStreamOnTable(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] Starting stream import")
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	v, err := client.Streams.ShowByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if err := d.Set("name", id.Name()); err != nil {
+		return nil, err
+	}
+	if err := d.Set("database", id.DatabaseName()); err != nil {
+		return nil, err
+	}
+	if err := d.Set("schema", id.SchemaName()); err != nil {
+		return nil, err
+	}
+	if err := d.Set("append_only", booleanStringFromBool(*v.Mode == sdk.StreamModeAppendOnly)); err != nil {
```

Comment:
nit: It's repeated a few times and could be `v.IsAppendOnly()`

Reply:
Done.